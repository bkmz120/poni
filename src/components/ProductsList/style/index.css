.productsList {
  display: flex;
  flex-wrap: wrap;
  padding-top: 20px;
}

.notFound {
  margin-top: 70px;
  text-align: center;
  font-size: 20px;
  color: #999;
}
