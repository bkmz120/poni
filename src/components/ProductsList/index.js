import React from 'react'
import PropTypes from 'prop-types'

import ProductItem from '@components/ProductItem'
import style from './style/index.css'

export default class ProductsList extends React.Component {
  render () {
    const {
      items,
    } = this.props

    return <div className={style.productsList}>
      { items.length > 0 && items.map((item) => <ProductItem key={item.name} product={item} />)}
      { items.length === 0 && <div className={style.notFound}>Пони не найдены</div>}
    </div>
  }
}

ProductsList.propTypes = {
  items: PropTypes.array,
}
