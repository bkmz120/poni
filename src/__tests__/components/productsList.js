/* eslint-disable */

import ProductsList from '@components/ProductsList'

describe('ProductsList',()=>{
  const products = [
    {
      'name': 'Твайлайт',
      'color': 'Фиолетовый',
      'kind': 'Единорог',
      'price': 19.99,
      'is_new': false,
    },
    {
      'name': 'Абдула',
      'color': 'Синий',
      'kind': 'Аликорн',
      'price': 29.99,
      'is_new': true,
    },
    {
      'name': 'Иосиф',
      'color': 'Красный',
      'kind': 'Пегас',
      'price': 9.99,
      'is_new': false,
    },
    {
      'name': 'Саня',
      'color': 'Черный',
      'kind': 'Единорог',
      'price': 19.99,
      'is_new': false,
    },
    {
      'name': 'Ли Кук',
      'color': 'Фиолетовый',
      'kind': 'Единорог',
      'price': 29.99,
      'is_new': true,
    },
    {
      'name': 'Портман',
      'color': 'Синий',
      'kind': 'Земная пони',
      'price': 19.99,
      'is_new': false,
    },
    {
      'name': 'Тиша',
      'color': 'Фиолетовый',
      'kind': 'Пегас',
      'price': 9.99,
      'is_new': false,
    },
    {
      'name': 'Черный лукич',
      'color': 'Черный',
      'kind': 'Земная пони',
      'price': 9.99,
      'is_new': false,
    },
  ]

  it('check snapshot', () => {
    const productsListWrapper = shallow(
        <ProductsList
          items={products}
        />
    );
    expect(productsListWrapper).toMatchSnapshot()
  })
})

