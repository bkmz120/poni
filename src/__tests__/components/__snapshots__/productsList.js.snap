// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsList check snapshot 1`] = `
<div
  className="productsList"
>
  <ProductItem
    key="Твайлайт"
    product={
      Object {
        "color": "Фиолетовый",
        "is_new": false,
        "kind": "Единорог",
        "name": "Твайлайт",
        "price": 19.99,
      }
    }
  />
  <ProductItem
    key="Абдула"
    product={
      Object {
        "color": "Синий",
        "is_new": true,
        "kind": "Аликорн",
        "name": "Абдула",
        "price": 29.99,
      }
    }
  />
  <ProductItem
    key="Иосиф"
    product={
      Object {
        "color": "Красный",
        "is_new": false,
        "kind": "Пегас",
        "name": "Иосиф",
        "price": 9.99,
      }
    }
  />
  <ProductItem
    key="Саня"
    product={
      Object {
        "color": "Черный",
        "is_new": false,
        "kind": "Единорог",
        "name": "Саня",
        "price": 19.99,
      }
    }
  />
  <ProductItem
    key="Ли Кук"
    product={
      Object {
        "color": "Фиолетовый",
        "is_new": true,
        "kind": "Единорог",
        "name": "Ли Кук",
        "price": 29.99,
      }
    }
  />
  <ProductItem
    key="Портман"
    product={
      Object {
        "color": "Синий",
        "is_new": false,
        "kind": "Земная пони",
        "name": "Портман",
        "price": 19.99,
      }
    }
  />
  <ProductItem
    key="Тиша"
    product={
      Object {
        "color": "Фиолетовый",
        "is_new": false,
        "kind": "Пегас",
        "name": "Тиша",
        "price": 9.99,
      }
    }
  />
  <ProductItem
    key="Черный лукич"
    product={
      Object {
        "color": "Черный",
        "is_new": false,
        "kind": "Земная пони",
        "name": "Черный лукич",
        "price": 9.99,
      }
    }
  />
</div>
`;
